tests/integration/targets/ec2_placement_group/tasks/main.yml

```diff
@@ -0,0 +1,218 @@
+- name: run ec2_lc tests
+  module_defaults:
+    group/aws:
+      aws_access_key: "{{ aws_access_key }}"
+      aws_secret_key: "{{ aws_secret_key }}"
+      security_token: "{{ security_token | default(omit) }}"
+      region: "{{ aws_region }}"
+  collections:
+    - amazon.aws
+
+  block:
+
+    - name: set up environment for testing.
+      include_tasks: env_setup.yml
+
+    - name: Create a placement group 1
+      community.aws.ec2_placement_group:
+        name: '{{ resource_prefix }}-pg1'
+        state: present
+      register: pg_1_create
+
+    - assert:
+        that:
+          - pg_1_create is changed
+          - pg_1_create.placement_group.name == '{{ resource_prefix }}-pg1'
+          - pg_1_create.placement_group.state == "available"
+          - '"ec2:CreatePlacementGroup" in pg_1_create.resource_actions'
+
+    - name: Gather information about placement group 1
+      community.aws.ec2_placement_group_info:
+        names:
+          - '{{ resource_prefix }}-pg1'
+      register: pg_1_info_result
+
+    - assert:
+        that:
+          - pg_1_info_result is not changed
+          - pg_1_info_result.placement_groups[0].name == '{{ resource_prefix }}-pg1'
+          - pg_1_info_result.placement_groups[0].state == "available"
+          - pg_1_info_result.placement_groups[0].strategy == "cluster"
+          - '"ec2:DescribePlacementGroups" in pg_1_info_result.resource_actions'
+
+    - name: Create a placement group 1 - Idempotency
+      community.aws.ec2_placement_group:
+        name: '{{ resource_prefix }}-pg1'
+        state: present
+      register: pg_1_create
+
+    - assert:
+        that:
+          - pg_1_create is not changed
+          - pg_1_create.placement_group.name == '{{ resource_prefix }}-pg1'
+          - pg_1_create.placement_group.state == "available"
+          - '"ec2:CreatePlacementGroup" not in pg_1_create.resource_actions'
+
+    - name: Create a placement group 2 with spread strategy
+      community.aws.ec2_placement_group:
+        name: '{{ resource_prefix }}-pg2'
+        state: present
+        strategy: spread
+      register: pg_2_create
+
+    - assert:
+        that:
+          - pg_2_create is changed
+          - pg_2_create.placement_group.name == '{{ resource_prefix }}-pg2'
+          - pg_2_create.placement_group.state == "available"
+          - '"ec2:CreatePlacementGroup" in pg_2_create.resource_actions'
```

Comment:
I n my opinion, you can remove all the statements similar to this `- '"ec2:CreatePlacementGroup" in pg_2_create.resource_actions'`

Reply:
@alinabuzachis I personally think it can be a decent additional check to make sure resource action took place.

Reply:
Ok, let's keep then.